configs/components/oifs/oifs.env.yaml

```diff
@@ -321,8 +321,8 @@ compiletime_environment_changes:
         LAPACK_LIB_DEFAULT: '"-L$MKLROOT/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_sequential"'
         # oasis
         OIFS_OASIS_BASE: '$(pwd)/oasis'
-        OIFS_OASIS_INCLUDE: '"-I$OIFS_OASIS_BASE/build/lib/psmile -I$OIFS_OASIS_BASE/build/lib/psmile/scrip -I$OIFS_OASIS_BASE/build/lib/psmile/mct -I$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu"'
-        OIFS_OASIS_LIB: '"-L$OIFS_OASIS_BASE/build/lib/psmile -L$OIFS_OASIS_BASE/build/lib/psmile/scrip -L$OIFS_OASIS_BASE/build/lib/psmile/mct -L$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -lpsmile -lmct -lmpeu -lscrip"'
+        OIFS_OASIS_INCLUDE: '"-I$OIFS_OASIS_BASE/build/lib/psmile -I$OIFS_OASIS_BASE/build/lib/psmile/scrip -I$OIFS_OASIS_BASE/build/lib/psmile/mct -I$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -I$OIFS_OASIS_BASE/lib/include/yac -I$OIFS_OASIS_BASE/lib/yaxt/include"'
+        OIFS_OASIS_LIB: '"-L$OIFS_OASIS_BASE/build/lib/psmile -L$OIFS_OASIS_BASE/build/lib/psmile/scrip -L$OIFS_OASIS_BASE/build/lib/psmile/mct -L$OIFS_OASIS_BASE/build/lib/psmile/mct/mpeu -L$OIFS_OASIS_BASE/lib/yac/lib -L$OIFS_OASIS_BASE/lib/yaxt/lib -lpsmile -lmct -lmpeu -lscrip -lyac_utils -lyac_core -lyaxt -lyaxt_c -qmkl"'
```

Comment:
Are this changes okay in terms of back-compatibility?

It's clearly having an effect in esm_tests (as it would be expected): https://github.com/esm-tools/esm_tools/actions/runs/8205605052/job/22442824281?pr=1127#step:8:65

Reply:
No, I think you found the part that was breaking the awicm3-v3.1.1 compile on this branch. I will need to develop a different solution!

Reply:
Won't this cause a compile error if I try to compile OpenIFS on Juwels without YAC, i.e. OASIS3-MCT4 or something?  
Could we instead make a variable `OIFS_OASIS_YAC_LIB = -lyac_utils -lyac_core -lyaxt -lyaxt_c` which we can add to `OIFS_OASIS_LIB` above? 
Then we can do something like this in `oasis.yaml`:

```python
choose_oasis.with_yac:
    1: 
        add_export_vars:
            OIFS_OASIS_YAC_LIB: "-lyac_utils -lyac_core -lyaxt -lyaxt_c"
    0:
        add_export_vars:
            OIFS_OASIS_YAC_LIB: ""
```

Reply:
That's a good suggestion. I will try this.

Reply:
In the oasis.3vn.yaml i added:
```
add_compiletime_environment_changes:
        choose_oasis3mct.with_yac:
                True:
                        OIFS_OASIS_YAC_LIBS: "-lyac_utils -lyac_core -lyaxt -lyaxt_c"
                        OIFS_OASIS_YAC_INC_PATH: "-I$OIFS_OASIS_BASE/lib/include/yac -I$OIFS_OASIS_BASE/lib/yaxt/include"
                        OIFS_OASIS_YAC_LIB_PATH: "-L$OIFS_OASIS_BASE/lib/yac/lib -L$OIFS_OASIS_BASE/lib/yaxt/lib"
                False:
                        OIFS_OASIS_YAC_LIBS: ""
                        OIFS_OASIS_YAC_INC_PATH: ""
                        OIFS_OASIS_YAC_LIB_PATH: ""
```
I get Key oasis3mct.with_yac was not defined. Same with `choose_with_yac`.

Reply:
Any advice on what I'm doing wrong here?

Reply:
Did you do something like this in `oasis3mct.yaml`:
```python
with_yac: False
...
choose_version:
   ...
   - '6.0-YAC':
       with_yac: True
```
or how is `with_yac` defined?

Reply:
Yes:

```
  6.0-YAC:
    # this version is taken directly from the CERFACS gitlab
    # it compiles in the standard way, i.e. not with cmake
    with_yac: True
```

Reply:
When people are back from holiday, I would like to have some help on this. I've been doing ugly workarounds for now. @mandresm @pgierz

Reply:
should be sorted now after the last commit